lightning/src/ln/onion_utils.rs

```diff
@@ -635,11 +642,23 @@ where L::Target: Logger {
 		}
 	}).expect("Route that we sent via spontaneously grew invalid keys in the middle of it?");
 	if let Some((channel_update, short_channel_id, payment_retryable)) = res {
-		(channel_update, short_channel_id, payment_retryable, error_code_ret, error_packet_ret)
+		DecodedOnionFailure {
+			network_update: channel_update, short_channel_id, payment_retryable,
```

Comment:
Given you are touching this, could you rename `channel_update` here to `network_update`?